src/tool_parsecfg.c

```diff
@@ -109,12 +109,12 @@ int parseconfig(const char *filename, struct GlobalConfig *global)
 #endif
   }
 
-  if(file != NULL)  /* WIN32: no need to fopen() again */
-    ;
-  else if(strcmp(filename, "-"))
-    file = fopen(filename, FOPEN_READTEXT);
-  else
-    file = stdin;
+  if(!file) { /* WIN32: no need to fopen() again */
```

Comment:
The Coverity warning occurs because it runs the check without `WIN32` defined.

So, with your additional check here for `!file` won't coverity now just spot an unncessary check since file is set to `NULL` unconditionally for non-WIN32 systems? I can't see how this change actually does anything else than the existing code.

My take was to just tell Coverity to ignore this since it triggers it based on an incomplete view of the situation.

Reply:
I ran this @captain-caveman2k's PR through coverity and it causes no new complaints and indeed fixes the existing one he set out to do!